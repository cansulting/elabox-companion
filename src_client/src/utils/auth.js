export const validCharacters = (str = "") => {
    if (!str || str.length <= 5) return false
    if (str.search(' ') >= 0) return false;

    const specialChars = /[`^$&()\'\"\[\]{};:\\|,.<>\/]/;
    return !specialChars.test(str);
}