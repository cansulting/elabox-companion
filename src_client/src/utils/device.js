import { ACCOUNT_PKID, AC_DEVICE_SERIAL, EboxEventInstance } from "../config";

export const getSerial = async () =>{
    try{
        const res = await EboxEventInstance.sendRPC(ACCOUNT_PKID, AC_DEVICE_SERIAL)
        if (res.code === 200) {
            // console.log("signin error", res)
            // throw res
            res.message = JSON.parse(res.message)
        }
        return res.message.Serial   
    }
    catch (error) {
        return error
    }
}