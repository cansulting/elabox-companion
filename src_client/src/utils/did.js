import { VerifiableCredential } from '@elastosfoundation/did-js-sdk';
import { connectivity, DID } from '@elastosfoundation/elastos-connectivity-sdk-js';
import { EssentialsConnector } from '@elabox/essentials-connector-client-browser';
let instance = null;
export default class Did {

    _initConnector() {
        console.log("initConnector")
        const connector = new EssentialsConnector()
        connectivity.registerConnector(connector)
        this.connector = connector
    }

    static getInstance() {
        if (!instance) {
            instance = new Did()
            instance._initConnector()
        }
        return instance 
    }

    async signin() {
        if (this.connector.hasWalletConnectSession())
            await this.connector.disconnectWalletConnect()
        const didAccess = new DID.DIDAccess()
        
        try {
            const presentation = await didAccess.requestCredentials(
                {claims: [DID.standardNameClaim("Activate elabox", false)]}
            );
            return presentation
        } catch (error) {
            console.log(error);
            return null
        }
        
    }

    disconnectConnector() {
        if (this.connector) {
            
            this.connector.disconnect()
            this.connector = null
        }
    }
}