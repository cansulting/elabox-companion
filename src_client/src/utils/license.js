import backend from "../api/backend"
import DID from "./did"

// return true or false if activated
export const isActivated = async () => {
    const res = await backend.isElaboxActivated()
    //console.log(res)
    return res.activated
}

export const activateLicense = async () => {
    const did = await DID.getInstance().request()
    //const did = "asdfsadf"
    if (did && did.holder) {
        const holder = did.holder.toJSON()
        const res = await backend.activateElabox(holder)
        return res 
    } 
    return null
}