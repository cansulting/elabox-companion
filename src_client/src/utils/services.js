export const eidSettings={
    isDisabled: true,
}