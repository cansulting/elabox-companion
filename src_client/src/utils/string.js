export const shortifyHash = (hash) => {
  return hash.substring(0, 9) + "..." + hash.substring(54, 64);
};

export const capitalize = (s) => {
  if (typeof s !== 'string') return ''
  return s.charAt(0).toUpperCase() + s.slice(1)
}