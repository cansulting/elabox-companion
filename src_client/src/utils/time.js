export const formatTime = (timeStamp) => {
  if (formatTime === null) return "...";

  let formattedTime = new Intl.DateTimeFormat("en-US", {
    year: "numeric",
    month: "2-digit",
    day: "2-digit",
    hour: "2-digit",
    minute: "2-digit",
    second: "2-digit",
  }).format(new Date(timeStamp * 1000));

  return formattedTime;
};
