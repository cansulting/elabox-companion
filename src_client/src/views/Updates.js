import React from "react"
import {
  Row,
  Col,
  Card,
  CardHeader,
  CardBody,
  Button,
  Progress,
  Spinner,
} from "reactstrap"
export default function Updates({ isMobile, ota }) {
  const {
    latestVersionDetails,
    hasNewUpdates,
    hasNewDownload,
    isProcessingData,
    isUpdated,
    notUpdated,
    isUpdating,
    isDownloading,
    progress,
  } = ota
  let body = ""
  let btnLabel = ""
  let headerLabel = ""
  const showNewDownload =
    hasNewDownload ||
    hasNewUpdates ||
    isDownloading ||
    isUpdating ||
    isUpdated ||
    notUpdated
  if (hasNewDownload) {
    body = (
      <div>
        <p>Name: {latestVersionDetails.name}</p>
        <p>Version: {latestVersionDetails.version}</p>
        <p>{latestVersionDetails.description}</p>
      </div>
    )
    headerLabel = "New Update available"
    btnLabel = "Download"
  } else if (isProcessingData) {
    body = (
      <Progress
        style={{ height: "50px" }}
        animated
        color="info"
        value={progress.percent}
      >
        {progress.status}
      </Progress>
    )
    headerLabel = "Processing...."
    btnLabel = <Spinner size="sm" color="light" children={""} />
  } else if (hasNewUpdates) {
    body = (
      <div>
        <p>Name: {latestVersionDetails.name}</p>
        <p>Version: {latestVersionDetails.version}</p>
        <p>{latestVersionDetails.description}</p>
      </div>
    )
    headerLabel = "Restart now"
    btnLabel = "Update"
  } else if (isUpdated) {
    body = (
      <Progress
        style={{ height: "50px" }}
        animated
        color="success"
        value={progress.percent}
      >
        {progress.status}
      </Progress>
    )
    headerLabel = "Update installed"
    btnLabel = "Ok"
  } else if (notUpdated) {
    body = (
      <Progress
        style={{ height: "50px" }}
        animated
        color="danger"
        value={progress.percent}
      >
        {progress.status}
      </Progress>
    )
    headerLabel = "Not updated"
    btnLabel = "Ok"
  }
  return (
    <div
      style={{
        ...{
          paddingLeft: "18%",
          width: "100%",
          backgroundColor: "#1E1E26",
        },
        ...(isMobile && { paddingLeft: undefined }),
      }}
      className="animated fadeIn w3-container"
    >
      {/* <Row>
        <Col>
          <Card
            style={{
              backgroundColor: "#272A3D",
              color: "white",
              fontSize: "16px",
              marginBottom: "20px",
            }}
          >
            <CardHeader>Installed Version</CardHeader>
            <CardBody>
              <p>Name: {currentVersionDetails.name}</p>
              <p>Version: {currentVersionDetails.version}</p>
              <p>{currentVersionDetails.description}</p>
            </CardBody>
          </Card>
        </Col>
      </Row> */}
      <Row>
        <Col>
          <Card
            style={{
              backgroundColor: "#272A3D",
              color: "white",
              fontSize: "16px",
              marginBottom: "20px",
            }}
          >
            <CardHeader>{headerLabel}</CardHeader>
            <CardBody>
              {showNewDownload ? (
                <Row>
                  <Col>{body}</Col>
                  {!isProcessingData && (
                    <Col
                      className="d-flex justify-content-center flex-column"
                      xs="2"
                    >
                      <Button
                        color="success"
                        onClick={() => {
                          if (hasNewDownload) {
                            ota.handleDownloadPackage()
                          } else if (hasNewUpdates) {
                            ota.handleUpdates()
                          } else if (isUpdated) {
                            window.location.reload()
                          }
                        }}
                        disabled={ota.disabledButton}
                      >
                        {btnLabel}
                      </Button>
                    </Col>
                  )}
                </Row>
              ) : (
                <h2 className="text-center">No updates</h2>
              )}
            </CardBody>
          </Card>
        </Col>
      </Row>
    </div>
  )
}
