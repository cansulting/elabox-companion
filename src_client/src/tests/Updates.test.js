import { fireEvent, screen,waitFor } from "@testing-library/react"
import { server, rest, BASE_URL, renderApp,clearLocalStorage,renderAsFragment } from "../utils/testing"
import Updates from "../views/Updates"
beforeAll(() => server.listen())
afterEach(() => server.resetHandlers())
afterAll(() => server.close())

describe("Updates",()=>{
    test("renders correctly",()=>{
        const asFragment=renderAsFragment(Updates)
        const html=asFragment()
        expect(html).toMatchSnapshot()
    })    
    describe("new updates",()=>{
        test("Able to download",async ()=>{
            renderApp()
            await screen.findByText("Sign In")
            const passwordInput = screen.getByTestId("password")
            const signInBtn = screen.getByTestId("sign-in-btn")
            fireEvent.change(passwordInput, { target: { value: "Tester" } })
            fireEvent.click(signInBtn)                
            const UpdatesElement=await screen.findByText("Updates")
            fireEvent.click(UpdatesElement)    
            await screen.findByText("New Update available")
            const downloadBtn=screen.getByTestId("download-btn")
            fireEvent.click(downloadBtn)
            const updateProcess=await screen.findByTestId("update-progress")
            expect(updateProcess).toBeInTheDocument()
        })
    })
    test("no updates",async ()=>{
        server.use(rest.get(`${BASE_URL}/check_new_updates`, (req, res, ctx) => {
            return res(
              ctx.json({
                current: 3,
                latest: 3,
                new_update: false,
                count: 1,
              })
            )
          }))
        renderApp()
        await screen.findByText("Sign In")
        const passwordInput = screen.getByTestId("password")
        const signInBtn = screen.getByTestId("sign-in-btn")
        fireEvent.change(passwordInput, { target: { value: "Tester" } })
        fireEvent.click(signInBtn)                
        const UpdatesElement=await screen.findByText("Updates")
        fireEvent.click(UpdatesElement)   
        expect(screen.queryByText("No updates")).not.toBeNull()
        expect(screen.queryByTestId("download-btn")).toBeNull()
    })
})