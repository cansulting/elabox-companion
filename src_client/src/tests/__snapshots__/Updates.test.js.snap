// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Updates renders correctly 1`] = `
<DocumentFragment>
  <component />
</DocumentFragment>
`;
